Diary/Controller/EditDiaryViewController.swift

```diff
@@ -0,0 +1,135 @@
+                                                               message: nil,
+                                                               preferredStyle: .actionSheet)
+        let shareAction: UIAlertAction = UIAlertAction(title: "Share...",
+                                                       style: .default) { _ in
+            var objectsToShare: [String] = []
+            objectsToShare.append(self.diaryModel.title + "\n" + self.diaryModel.body)
+            self.showActivityContoller(objectsToShare)
+        }
+        
+        let deleteAction: UIAlertAction = UIAlertAction(title: "Delete",
+                                                        style: .destructive) { _ in
+            let alert: UIAlertController = UIAlertController(title: "진짜요?",
+                                                             message: "정말로 삭제하시겠어요?",
+                                                             preferredStyle: .alert)
+            let cancelAction: UIAlertAction = UIAlertAction(title: "취소", style: .cancel)
+            let deleteAction: UIAlertAction = UIAlertAction(title: "삭제", style: .destructive) { _ in
+                do {
+                    try CoreDataMananger.shared.deleteDiary(self.diaryModel)
+                } catch {
+                    self.present(ErrorAlert.shared.showErrorAlert(title: DiaryError.deleteFailed.alertTitle,
+                                                                  message: DiaryError.deleteFailed.alertMessage,
+                                                                  actionTitle: "확인"),
+                                 animated: true)
+                }
+                
+                self.navigationController?.popViewController(animated: true)
+            }
+            
+            alert.addAction(cancelAction)
+            alert.addAction(deleteAction)
+            self.present(alert, animated: true)
+        }
+        
+        let cancelAction: UIAlertAction = UIAlertAction(title: "Cancel",
+                                                        style: .cancel)
+        
+        actionSheet.addAction(shareAction)
+        actionSheet.addAction(deleteAction)
+        actionSheet.addAction(cancelAction)
+        present(actionSheet, animated: true)
+    }
+    
+    private func setKeyboardHideObserver() {
+        NotificationCenter.default.addObserver(self,
+                                               selector: #selector(self.keyboardDidHide),
+                                               name: UIResponder.keyboardDidHideNotification,
+                                               object: nil)
+    }
+    
+    @objc private func keyboardDidHide() {
+        self.updateCurrentDiary()
+    }
+    
+    func updateCurrentDiary() {
+        do {
+            let diaryContent = self.editDiaryView.fetchTextViewContent()
+            
+            try CoreDataMananger.shared.updateDiary(self.createDiaryModel(with: diaryContent))
```

Comment:
```suggestion
            let newDiary: DiaryModel = makeNewDiary(with: diaryContent)
            try CoreDataMananger.shared.update(diary: newDiary)
```

이런 식으로 써진다면 더 읽기 쉬워질거 같아요!

Reply:
제안해주신 대로 수정했습니다!!